### DISCUSSION TITLE
GTK Apprt: Keyboard shortcuts dialog

### DISCUSSION THREAD

**Discussion by anmolw** [ghostty-org/ghostty#3863] [Feature Requests, Ideas]
It would be nice to have a keyboard shortcuts dialog for the GTK Apprt, similar to that of Gnome Console or Black Box. Ideally this would take into account configuration modifications made by the user. Having this in place would help with discoverability, especially for those unaware of the `+list-keybinds` CLI action.

For reference, this is what Gnome Console's dialog looks like:

![image](https://github.com/user-attachments/assets/73a4d753-6004-4618-bc69-e37068b94875)

cc @tristan957 @jcollie

**Comment by pluiedev** [comment]
Related to (or even a part of) #2354.

**Comment by pluiedev** [comment]
From a UI design perspective I don't actually like this layout. For starters, the keybind window would need to be quite big relative to the terminal window itself, which could be quite overwhelming. Secondly, we have parametrized actions, and the parameters can be hard to express in this style of dialog. Lastly, we have a lot of categories of keybinds, and not all are relevant at all times and they don't need to be shown at all times.

I think a better design would be the keybinds dialog like GNOME Builder with foldable categories and a simple list of vertical entries - that way it's far more compact and also easier on organization.
